lib/src/models/kyc_requirement.dart

```diff
@@ -0,0 +1,164 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:kyc_client_dart/kyc_client_dart.dart';
+import 'package:kyc_client_dart/src/api/models/v1_data_type.dart';
+import 'package:kyc_client_dart/src/api/models/v1_document_field_type.dart';
+import 'package:kyc_client_dart/src/api/models/v1_document_type.dart';
+import 'package:kyc_client_dart/src/api/models/v1_formula.dart';
+import 'package:kyc_client_dart/src/api/models/v1_get_kyc_requirements_response.dart';
+
+part 'kyc_requirement.freezed.dart';
+part 'kyc_requirement.g.dart';
+
+enum BasicInfoType {
+  email,
+  phone,
+  selfie,
+}
+
+enum DocumentField {
+  idNumber,
+  photoFront,
+  photoBack,
+}
+
+@freezed
+class KycRequirement with _$KycRequirement {
+  const factory KycRequirement({
+    required String country,
+    required List<Requirement> requirements,
+  }) = _KycRequirement;
+
+  factory KycRequirement.fromProto(V1GetKycRequirementsResponse response) {
+    final country = response.country;
+    final requirements = <Requirement>[];
+
+    for (final req in response.requirements) {
+      final type = req.type;
+      final formula = req.formula;
+
+      if (type == V1DataType.dataTypeEmail) {
+        requirements
+            .add(const Requirement.basicInfo(type: BasicInfoType.email));
+      } else if (type == V1DataType.dataTypePhone) {
+        requirements
+            .add(const Requirement.basicInfo(type: BasicInfoType.phone));
+      } else if (type == V1DataType.dataTypeSelfieImage) {
+        requirements
+            .add(const Requirement.basicInfo(type: BasicInfoType.selfie));
+      } else if (type == V1DataType.dataTypeDocument) {
+        final docReq = _parseFormula(formula);
+        if (docReq != null) {
+          requirements.add(docReq);
+        }
+      }
+    }
+
+    return KycRequirement(
+      country: country,
+      requirements: requirements,
+    );
+  }
+
+  static Requirement? _parseFormula(V1Formula formula) {
```

Comment:
Here as well, `_parseFormula` should return `Requirement` and throw if something goes wrong.

Reply:
updated code